[IMP] Inventory: Update product type documentation (#11409)
odoo/documentation: content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst

=====================

Products and services can be designated as those that can be bought, sold, or both. On the product
form, tick the :guilabel:`Sales` checkbox if a product can be *sold* to a customer (e.g. finished
goods). Tick :guilabel:`Purchase` if the product can be *purchased* (e.g. raw materials).

.. example::
  If a resale clothing shop buys discounted denim jackets and sells them at a higher cost to the
  end consumer, the `Jacket` product form might have *both* the :guilabel:`Sales` and
  :guilabel:`Purchase` box ticked.

  On the other hand, let's say the store occasionally sews new jackets using denim and thread as raw
  materials. In the `Denim` and `Thread` product forms, only :guilabel:`Purchase` should be ticked,
  whereas the `Handmade Jacket` product form would only tick :guilabel:`Sales`.

Goods vs. services
==================

:guilabel:`Product Type` needs to be selected on the :guilabel:`General Information` tab. Each
product type impacts different operations in other Odoo applications, such as *Sales* and
*Purchasing*, and should be chosen carefully.

- :guilabel:`Goods`: A tangible, material object (e.g. anything from a hamburger to a house).

> - :guilabel:`Service`: An intangible, immaterial offering (e.g. a repair, a haircut, call center

Author: Love this- how did I get through grad school without any of my professors telling me this??  :)

  included (*service*)).

.. note::
  Due to their immaterial nature, services are not trackable in Odoo's *Inventory* application.

.. _inventory/product_management/manufacture:

Configuring goods
=================

Selecting :guilabel:`Goods` as the product type automatically triggers the appearance of a few
fields in Odoo:

- :guilabel:`Inventory` tab: From here,
  :doc:`purchasing and manufacturing routes <../../shipping_receiving/daily_operations/use_routes>`
  and product logistics (such as product weight and customer lead time) can be specified.

- :ref:`Invoicing Policy <inventory/product_management/invoicing-policy>` field.

.. note::
   The :guilabel:`Invoicing Policy` field will only appear if the *Sales* app is installed.

- :ref:`Track Inventory <inventory/product_management/tracking-inventory>` field.

- Smart buttons: Some smart buttons are triggered right away when :guilabel:`Goods` is selected;
  others show upon additionally selecting a :guilabel:`Track Inventory` method. For example,
  :guilabel:`On Hand` and :guilabel:`Forecasted` will display when :guilabel:`Track Inventory` is

>   ticked.

Author: I see why you say this and would agree, but I enumerate the most important smart buttons (for inventory, since this is an inventory doc) later in this doc. So instead, I am adding the following to point users in the right direction: "In general, the smart buttons on a product form help execute :ref:`inventory operations
  <_inventory/product_management/inventory-ops-by-product-type>.`"

.. image:: type/product-form.png
  :alt: Designate a product as a good or service.

.. _inventory/product_management/invoicing-policy:

Invoicing policy
----------------

The :guilabel:`Invoicing policy` field only shows on the product form if a product is for sale (in
other words, if :guilabel:`Sales` is ticked, and the *Sales* app is installed).

When configuring a product for sale, it is necessary to choose an
:doc:`invoicing policy <../../../../sales/sales/invoicing/invoicing_policy>`. When an invoicing
policy of :guilabel:`Ordered quantities` is selected, customers are invoiced once the sales order
is confirmed. When :guilabel:`Delivered quantities` is ticked, customers are invoiced once the
delivery is completed.

.. _inventory/product_management/tracking-inventory:

Tracked vs. untracked goods
---------------------------

The :guilabel:`Track Inventory` field on the product form determines a lot of Odoo's *Inventory*
operations.

- **Tracked** products are those for which stock and inventory are maintained. Examples include

>   finished goods and, often, the raw materials or components needed to make them.

Author: I think, since components aren't necessarily raw materials, it doesn't need to be a parenthetical.

  When :guilabel:`Track Inventory` is ticked, a dropdown menu appears, offering for inventory to be
  tracked one of three ways: :guilabel:`By Unique Serial Number`, :guilabel:`By Lots`, and

>   :guilabel:`By Quantity`.

Author: I think if it's standing alone, it would be a dropdown, but if it's a modifier (as it is in this case), it should be hyphenated?

  Click the button to view or add stock levels for a tracked product.

- :icon:`fa-chart` :guilabel:`Forecasted`: Number of units *expected* to be available in inventory
  after all orders are taken into account. In other words,
  `forecasted = on hand quantity + incoming shipments - outgoing shipments`. Click the button
  to view the :guilabel:`Forecasted Report`.

On the other hand, untracked products are regarded as *always* available. Consequently,
:guilabel:`On-Hand Quantity` is not tracked, and there is no `Forecasted` quantity available.

.. _inventory/product_management/putaway:

Putaway rules & storage
***********************

Both tracked and untracked goods can optimize storage using:

- :icon:`fa-random` :doc:`Putaway Rules <../../shipping_receiving/daily_operations/putaway>`:
  Putaway rules that apply to a good, such as where to store it when a new shipment arrives.

- :icon:`fa-cubes`
  :doc:`Storage Capacities <../../shipping_receiving/daily_operations/storage_category>`: Any
  storage capacity limitations specified for this good. For example, a warehouse may require that

>   only 10 (or less) sofas be stored there at any given time, due to their large size.

Reviewer: ```suggestion
  only ten (or less) sofas be stored there at any given time, due to their large size.
```

appear for manufacturing operations:

- :icon:`fa-flask` :guilabel:`Bill of Materials`: The BoMs used to make this product.

- :icon:`fa-level-up` :guilabel:`Used In`: Other goods that include this product in their BoM.

.. _inventory/product_management/transfer-store:

Transferring goods
~~~~~~~~~~~~~~~~~~

*Transfers* are warehouse operations that involve the movement of goods. Examples of transfers
include
:doc:`deliveries and receipts <../../shipping_receiving/daily_operations/receipts_delivery_one_step>`
as well as :doc:`internal transfers <../../warehouses_storage/replenishment/resupply_warehouses>`
between warehouses.

When creating a transfer for tracked products in the *Inventory* app, transfers modify the on-hand
quantity at each location. For example, transferring five units from the internal location
`WH/Stock` to `WH/Packing Zone` decreases the recorded quantity at `WH/Stock` and increases it at
`WH/Packing Zone`.

For untracked products, transfers can be created, but exact quantities at each storage location are
not tracked.

.. _inventory/product_management/package:

Packages
~~~~~~~~

> Both tracked and untracked, or non-inventory, products can be placed in :doc:`packages <package>`.

Reviewer: ```suggestion
Both tracked and untracked (non-inventory) products can be placed in :doc:`packages <package>`.
```

However, for non-inventory products, the quantity is not tracked, and the product is not listed in
the package's :guilabel:`Contents` (which can be accessed by going to :menuselection:`Inventory app
--> Products --> Packages`, and selecting the desired package).

.. figure:: type/package-content.png
   :align: center
   :alt: Show Packages page, containing the package contents list.

   An untracked product was placed in the package, but the **Content** section does not list it.

> Additionally, if the :guilabel:`Move Entire Packages` feature is enabled

Reviewer: ```suggestion
Additionally, if the *Move Entire Packages* feature is enabled
```

> (:menuselection:`Inventory --> Configuration --> Operations Types`, and select any operation),

Reviewer: ```suggestion
(:menuselection:`Inventory app --> Configuration --> Operations Types`, select any operation, and tick the :guilabel:`Move Entire Packages` checkbox),
```

> moving a package updates the location of the contained tracked products but not the contained

Reviewer: ```suggestion
moving a package updates the location of the contained tracked products, but not the contained
```

Author: Comma not needed because it's not connecting two independent clauses

untracked products.

.. _inventory/product_management/report:

Inventory reports
~~~~~~~~~~~~~~~~~

**Only** tracked products appear on the following reports. Note, these reports are only available

> to users with :doc:`administrator access <../../../../general/users/access_rights>`.

Reviewer: turn the 'Note' sentence into an admonition

- :doc:`Stock report <../../warehouses_storage/reporting/stock>`: A comprehensive list of all
  on-hand, unreserved, incoming, and outgoing tracked inventory. To access the report, go to
  :menuselection:`Inventory app --> Reporting --> Stock`.

- :doc:`Location report  <../../warehouses_storage/reporting/locations>`: A breakdown of which
  tracked products are held at each location (internal, external, or virtual). The report is only
  available with the *Storage Location* feature activated (:menuselection:`Inventory app -->
  Configuration --> Settings`). To access the report, go to :menuselection:`Inventory app -->
  Reporting --> Locations`.

- :doc:`Moves History report <../../warehouses_storage/reporting/moves_history>`: An overview of
  where and when this good has moved in/out of stock. To access the report, go to
  :menuselection:`Inventory app --> Reporting --> Moves History`. Alternatively, click the
  :icon:`fa-exchange` :guilabel:`In / Out` smart button on a product form to filter the report

>   on that product's moves history.

Reviewer: ```suggestion
  on that product's specific moves history.
```

> - :guilabel:`Moves Analysis`: A pivot table view of inventory transfers by operation type.

Reviewer: ```suggestion
- :guilabel:`Moves Analysis`: This report provides a pivot table view of inventory transfers by operation type.
```

> 

Reviewer: ```suggestion
```

> - :ref:`Stock Valuation report <inventory/management/reporting/valuation-report>`: A detailed record

Reviewer: ```suggestion
- :ref:`Stock Valuation report <inventory/management/reporting/valuation-report>`: This report provides a detailed record
```

==================

When configuring a product, a :guilabel:`Product Type` needs to be selected on the
:guilabel:`General Information` tab of a product form. Each product type impacts different
operations in other Odoo applications, such as **Sales** and **Purchase**, and should be chosen
carefully.

- :guilabel:`Goods`: a tangible, material object (e.g. anything from a hamburger to a house)
- :guilabel:`Service`: an intangible, immaterial offering (e.g., a repair, a haircut, call center
  assistance)
- :guilabel:`Combo`: any mix of goods and services (e.g. a new car (*good*) with an oil change
  included (*service*))

.. note::
  Due to their immaterial nature, services are not trackable in Odoo's **Inventory** application.

.. _inventory/product_management/manufacture:

Configure goods
===============

Selecting :guilabel:`Goods` as the :guilabel:`Product Type` automatically triggers the appearance of
a few fields and tabs in the product form:

- :guilabel:`Inventory` tab: From here,
  :doc:`purchasing and manufacturing routes <../../shipping_receiving/daily_operations/use_routes>`
  and product logistics, such as product weight and customer lead time, can be specified.

> - :ref:`Invoicing Policy <inventory/product_management/invoicing-policy>` field: This field
>   determines at what point in the sales process a customer is invoiced.
> 
> .. important::
>    The :guilabel:`Invoicing Policy` field **only** appears if the **Sales** app is installed.
> 

Reviewer 2: Nitpick, please indent admonitions to their relevant bullet points :)
```suggestion
- :ref:`Invoicing Policy <inventory/product_management/invoicing-policy>` field: This field
  determines at what point in the sales process a customer is invoiced.

  .. important::
     The :guilabel:`Invoicing Policy` field **only** appears if the **Sales** app is installed.

```

     - Yes
     - No
   * - :doc:`Use inventory valuation <../inventory_valuation/using_inventory_valuation>`
     - Yes
     - No
   * - :ref:`Create transfer <inventory/product_management/transfer-store>`
     - Yes
     - Yes
   * - :doc:`Use lot/serial number tracking <../product_tracking>`
     - Yes
     - No
   * - :doc:`Can be placed in a kit <../../../manufacturing/advanced_configuration/kit_shipping>`
     - Yes
     - Yes
   * - :ref:`Can be placed in a package <inventory/product_management/package>`
     - Yes
     - Yes
   * - :ref:`Appears on inventory reports <inventory/product_management/report>`
     - Yes
     - No

Inventory
~~~~~~~~~

.. _inventory/product_management/on-hand:

On-hand and forecasted quantities
*********************************

A tracked product's on-hand and forecasted quantities, based on incoming and outgoing orders, are
reflected on the product form with two smart buttons:

- :icon:`fa-cubes` :guilabel:`On-Hand Quantity`: This represents the number of units currently
  available in inventory. Click the button to view or add stock levels for a tracked product.

> - :icon:`fa-chart` :guilabel:`Forecasted`: This represents the number of units *expected* to be

Reviewer 2: Update icon class :)
```suggestion
- :icon:`fa-area-chart` :guilabel:`Forecasted`: This represents the number of units *expected* to be
```

  available in inventory after all orders are taken into account. In other words,

>   `forecasted = on hand quantity + incoming shipments - outgoing shipments`. Click the button
>   to view the :guilabel:`Forecasted Report`.

Reviewer 2: Fun fact! Totally optional to use the [:math: role](https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#math) for maths!

(sorry for some reason this suggestion was not added to my last review)

<img width="960" alt="Screenshot 2024-12-03 at 5 57 28 PM" src="https://github.com/user-attachments/assets/63df8b43-5920-4d2e-847b-aed1b4e5ec32">

```suggestion
  :math:`\text{forecasted} = \text{on hand quantity} + \text{incoming shipments} -
  \text{outgoing shipments}`. Click the button to view the :guilabel:`Forecasted Report`.
```

include
:doc:`deliveries and receipts <../../shipping_receiving/daily_operations/receipts_delivery_one_step>`,
as well as :doc:`internal transfers <../../warehouses_storage/replenishment/resupply_warehouses>`
between warehouses.

When creating a transfer for tracked products in the **Inventory** app, transfers modify the on-hand
quantity at each location. For example, transferring five units from the internal location
`WH/Stock` to `WH/Packing Zone` decreases the recorded quantity at `WH/Stock` and increases it at
`WH/Packing Zone`.

For untracked products, transfers can be created, but exact quantities at each storage location are
not tracked.

.. _inventory/product_management/package:

Packages
~~~~~~~~

Both tracked and untracked (non-inventory), products can be placed in :doc:`packages <package>`.

However, for non-inventory products, the quantity is not tracked, and the product is not listed in
the package's :guilabel:`Contents` (which can be accessed by going to :menuselection:`Inventory app
--> Products --> Packages`, and selecting the desired package).

.. figure:: type/package-content.png

>    :align: center

Reviewer 2: Remove center alignment attribute
```suggestion
```

available.

.. _inventory/product_management/putaway:

Putaway rules and storage
*************************

Both tracked and untracked goods can optimize storage using:

- :icon:`fa-random` :doc:`Putaway Rules <../../shipping_receiving/daily_operations/putaway>`:
  This represents putaway rules that apply to a good, such as where to store it when a new shipment
  arrives.
- :icon:`fa-cubes`
  :doc:`Storage Capacities <../../shipping_receiving/daily_operations/storage_category>`:
  This represents any storage capacity limitations specified for this good. For example, a warehouse
  may require that only ten (or less) sofas be stored there at any given time, due to their large
  size.

.. _inventory/product_management/replenishment:

Replenishment
*************

Reordering rules
^^^^^^^^^^^^^^^^

Only tracked products can trigger
:doc:`reordering rules <../../warehouses_storage/replenishment/reordering_rules>` to generate
purchase orders. Untracked goods *cannot* be managed using reordering rules.

Reordering rules can be configured directly on the product form via the
:icon:`fa-refresh` :guilabel:`(refresh)` icon.

.. note::

>    If reordering rules already exist on a product, Odoo will relabel this button to

Reviewer: ```suggestion
   If reordering rules already exist on a product, Odoo re-labels this button to
```

   :guilabel:`Min / Max`, to show the minimum and maximum number of units that must be in stock.

.. _inventory/product_management/po:

Create purchase orders
^^^^^^^^^^^^^^^^^^^^^^

Both tracked and untracked products can be included in a request for quotation in the **Purchase**
app. However, when receiving untracked products, their on-hand quantity does not change upon
validating the receipt (`WH/IN`).

Replenish smart button
^^^^^^^^^^^^^^^^^^^^^^

The :guilabel:`Replenish` smart button allows all goods to be restocked directly from the product
form, according to the *Preferred Route*.

> .. seealso::
>   :doc:`Replenishment <../../warehouses_storage/replenishment>`
>   `Odoo Tutorials: Replenishment Methods for Manufacturing
>   <https://www.youtube.com/watch?v=vtjeMGcG8aM>`_

Reviewer: ```suggestion
.. seealso::
   :doc:`Replenishment <../../warehouses_storage/replenishment>`
   `Odoo Tutorials: Replenishment Methods for Manufacturing
   <https://www.youtube.com/watch?v=vtjeMGcG8aM>`_
```

.. _inventory/product_management/manufacturing:

Manufacturing
~~~~~~~~~~~~~

Both tracked and untracked products can be manufactured,
:doc:`subcontracted <../../../manufacturing/subcontracting>`, or included in another product's
:doc:`bill of materials (BoM) <../../../manufacturing/basic_setup/bill_configuration>`.

.. _inventory/product_management/BoM:

On the product form for a tracked or untracked good, there are several smart buttons that may
appear for manufacturing operations:

> - :icon:`fa-flask` :guilabel:`Bill of Materials`: This show the BoMs used to make this product.
> - :icon:`fa-level-up` :guilabel:`Used In`: This shows other goods that include this product in their
>   BoM.

Reviewer: ```suggestion
- :icon:`fa-flask` :guilabel:`Bill of Materials`: shows the BoMs used to make this product
- :icon:`fa-level-up` :guilabel:`Used In`: shows other goods that include this product in their
  BoM
```

Author: I think I should leave these in for consistency; you had suggested previously that, for the other bullet points introducing icons / reports, I use transition phrases like, "This respresents..." etc.

between warehouses.

When creating a transfer for tracked products in the **Inventory** app, transfers modify the on-hand
quantity at each location. For example, transferring five units from the internal location
`WH/Stock` to `WH/Packing Zone` decreases the recorded quantity at `WH/Stock` and increases it at
`WH/Packing Zone`.

For untracked products, transfers can be created, but exact quantities at each storage location are
not tracked.

.. _inventory/product_management/package:

Packages
~~~~~~~~

Both tracked and untracked (non-inventory), products can be placed in :doc:`packages <package>`.

However, for non-inventory products, the quantity is not tracked, and the product is not listed in
the package's :guilabel:`Contents` (which can be accessed by going to :menuselection:`Inventory app
--> Products --> Packages`, and selecting the desired package).

.. figure:: type/package-content.png
   :align: center
   :alt: Show Packages page, containing the package contents list.

   An untracked product was placed in the package, but the **Content** section does not list it.

Additionally, if the *Move Entire Packages* feature is enabled, moving a package updates the

> location of the contained tracked products but not the contained untracked products. To enable this

Reviewer: ```suggestion
location of the contained tracked products, but not the contained untracked products. To enable this
```

Author: Leaving out since it's not an independent clause

> feature, navigate to :menuselection:`Inventory --> Configuration --> Operations Types`, select any

Reviewer: ```suggestion
feature, navigate to :menuselection:`Inventory app --> Configuration --> Operations Types`, select any
```

operation, and tick the :guilabel:`Move Entire Packages` checkbox.

> 

Reviewer: ```suggestion
```
